// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/compiler/computedPropertiesNarrowed.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,20 +1,26 @@
 
 
 //// [computedPropertiesNarrowed.d.ts]
-export declare let o: invalid;
+declare const x: 0 | 1;
+export declare let o: {
+    [x]: number;
+};
 declare const y: 0;
 export declare let o2: {
-    0: number;
+    [y]: number;
 };
 export declare let o3: {
     1: number;
 };
 export declare let o31: {
     [-1]: number;
 };
 export declare let o32: invalid;
-export declare let o4: invalid;
+declare let u: invalid;
+export declare let o4: {
+    [u]: number;
+};
 export declare let o5: invalid;
 declare const uu: unique symbol;
 export declare let o6: {
     [uu]: number;
@@ -23,31 +29,28 @@
 declare let E: {
     readonly A: 1;
 };
 export declare const o8: {
-    1: number;
+    [E.A]: number;
 };
 export declare const o9: invalid;
 export {};
 
 /// [Errors] ////
 
-computedPropertiesNarrowed.ts(5,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 computedPropertiesNarrowed.ts(18,20): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
-computedPropertiesNarrowed.ts(22,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
+computedPropertiesNarrowed.ts(20,9): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 computedPropertiesNarrowed.ts(26,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 computedPropertiesNarrowed.ts(37,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 computedPropertiesNarrowed.ts(47,5): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 
 
-==== computedPropertiesNarrowed.ts (6 errors) ====
+==== computedPropertiesNarrowed.ts (5 errors) ====
     const x: 0 | 1 = Math.random()? 0: 1;
     declare function assert(n: number): asserts n is 1;
     assert(x);
     export let o = {
         [x]: 1 // error narrow type !== declared type
-        ~~~
-!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     }
     
     
     const y: 0 = 0
@@ -63,12 +66,12 @@
                        ~~~~~
 !!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     
     let u = Symbol();
+            ~~~~~~~~
+!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     export let o4 = {
         [u]: 1 // Should error, nut a unique symbol
-        ~~~
-!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     }
     
     export let o5  ={
         [Symbol()]: 1 // Should error
